pyogrio/geopandas.py

```diff
@@ -183,17 +183,19 @@ def read_dataframe(
     import pandas as pd
     import geopandas as gp
     from geopandas.array import from_wkb
+    import shapely  # if geopandas is present, shapely is expected to be present
 
     path_or_buffer = _stringify_path(path_or_buffer)
 
     read_func = read_arrow if use_arrow else read
+    force_2d_read = False if use_arrow else force_2d
```

Comment:
This could be `gdal_force_2d` for consistency with `gdal_skip_features` introduced for a roughly similar purpose in #282 , just as a signal that it is a parameter getting passed down to the Cython / GDAL level.